pkg/api/pss_test.go

```diff
@@ -348,37 +337,41 @@ func TestPssPingPong(t *testing.T) {
 
 	p.TryUnwrap(tc)
 
-	waitMessage(t, msgContent, nil, &mtx)
+	go readMessage(t, cl, msgC)
+	go expectMessage(t, respC, msgC, nil)
+	if err := <-respC; err != nil {
+		t.Fatal(err)
+	}
 }
 
-func waitReadMessage(t *testing.T, mtx *sync.Mutex, cl *websocket.Conn, targetContent []byte, done <-chan struct{}) {
+func readMessage(t *testing.T, cl *websocket.Conn, msgC chan []byte) {
```

Comment:
So not really sure why we need 2 go routines here. Both are governed by the same timer and seems like we can accomplish the same thing with just 1 routine no?

Reply:
It's possible, I have further modified code so one goroutine is created.